### ISSUE TITLE
Rumble sound before stage 1 boss

### ISSUE THREAD

**Issue by pepodmc3** [ArcadeTV/msu-md-tmnthsh#2]
There is a rumble sound between the end of "stage 1-3" track and the start of the "stage boss" track. But it doesnt sounds.

**Comment by pepodmc3** [comment]
As i wrote in twitter, take in consideration that the boss music starts after the earthquake ends, so it will be worse if i add the rumble on the track because the will seem as if the sound is late

And it doesnt sound in other bosses appearances.

I would let it as it is. The earthquake only last two seconds anyway.

**Comment by ArcadeTV** [comment]
The good news is that I can make the audio track start as soon as the rumble would start in the original game, so there will be no delay. We will have to decide if we want to add a new track to the soundPack. If yes, we take the boss music and duplicate it, adding rumble to the beginning. The other boss music is unaffected from this change.
Of course, the order of the tracks in the audio pack will change if we do this.

**Comment by pepodmc3** [comment]
> The good news is that I can make the audio track start as soon as the rumble would start in the original game, so there will be no delay. We will have to decide if we want to add a new track to the soundPack. If yes, we take the boss music and duplicate it, adding rumble to the beginning. The other boss music is unaffected from this change.
> Of course, the order of the tracks in the audio pack will change if we do this.

I dont want to sound lazy, but i would leave it as it is now (i mean, its only one sound of two seconds)

Plus, the boss track in the soundpack has a 2:40 minutes duration. 
As i extended the tracks, i had to reduce some of them (they are still extended but less) so some tracks didnt start late in other stages (a consecuence of the extended tracks, as it works like it worked in a Sega CD, seems that the sega cd takes longer to read some tracks depending of the position of the track in the track list, so i had to find a balance about how much i can extend them)
So, if we add 2:40 minutes more, i will need to reduce other tracks to find the balance again, and test all again.

**Comment by pepodmc3** [comment]
As an example, i was doing a extended music pack for sonic 1, but, even if all tracks starts when it should and all was practically perfect, the Drowning track starts very late, because its a track that starts from a sudden, and the extended tracks makes the sega cd spend a lot of time until it reach that track.

**Comment by pepodmc3** [comment]
Something that i was thinking was that, even if Krizz fix the loop problem in the driver, the Real hardware will work as it always work.

I mean, people using Mega SD will not have any problem. But people using real Sega CD hardware will be always with the full track loops , not with the "loops from whatever part of the track" that Krizz driver had implemented.

So , extended versions of the music packs will still be useful for some people.

People with everdrives still need to use Sega CD hardware for Sega CD games?

**Comment by ArcadeTV** [comment]
> People with everdrives still need to use Sega CD hardware for Sega CD games?

No, they work straight from the cart jut fine.
To compliment seek times, you'd want the tracks close to each other so the gap is minimized on real hardware.

**Comment by pepodmc3** [comment]
> > People with everdrives still need to use Sega CD hardware for Sega CD games?
> 
> No, they work straight from the cart jut fine.
> To compliment seek times, you'd want the tracks close to each other so the gap is minimized on real hardware.


But sometimes its complicated. In the sonic 1 case, labyrinth zone is the 2nd track and "Drowning" is the track 18. The two tracks are very far from each other, and drowning tracks starts from a sudden.

When you do the patchs for each game, can you modify the position of the track in a tracklist? like changing places of some tracks? (Put Drowning as the track 9 and put special stage as track 18)

**Comment by ArcadeTV** [comment]
Yes, this can be done freely. Also, it may be useful to put such tracks like "dying" or "drowning" multiple times on the disc after every track to relax the situation for a laser when skipping tracks.
Keep in mind that ALL existing soundpacks would have to reflect those changes!

**Comment by pepodmc3** [comment]
> Yes, this can be done freely. Also, it may be useful to put such tracks like "dying" or "drowning" multiple times on the disc after every track to relax the situation for a laser when skipping tracks.


Thats interesting.
This would be needed for games like sonic 1 that has tracks (invincibility,drowning) that interrupts stage tracks from a sudden.

But its very cool to know that you can duplicate tracks in a tracklist when making patches.

I finished the music pack done with the Rumble sound and the stage clear theme fixed.

**Comment by ArcadeTV** [comment]
I will create a "branch" on github for adding the new logic to the code and add the new tracks to the playlist.

**Comment by pepodmc3** [comment]
> I will create a "branch" on github for adding the new logic to the code and add the new tracks to the playlist.

Are you talking about the new "boss track" ?

**Comment by ArcadeTV** [comment]
yes. I will create a new branch in git and merge it when we are good with the changes.

**Comment by pepodmc3** [comment]
> yes. I will create a new branch in git and merge it when we are good with the changes.

I was looking to Sonic 1 guides , and only in two stages appears the power sneakers item (rapid shoes). In Green Hill and Spring Yard. So the rest of the "rapid versions" of the tracks are not needed when doing a music pack.

Im testing sonic 1 to see in which part i would need a duplication of a track. At the moment, it seems that the only change i would need would be the drowning track near labyrinth zone track (next to) so it starts quickly.

I would fix your music pack to reflect the change. But im still testing.
